weaviate/collections/classes/config_base.py

```diff
@@ -68,3 +69,12 @@ class _QuantizerConfigUpdate(_ConfigUpdateModel):
     @abstractmethod
     def quantizer_name() -> str:
         ...
+
+
+@dataclass
+class _EnumLikeStr:
+    string: str
+
+    @property
+    def value(self) -> str:
+        return self.string
```

Comment:
What's the motivation for this class? Why isn't a string enough, so `Union[Rerankers, str]`? Is it because you can't differentiate between the `StrEnum` and `str` easily when parsing to the Weaviate object?

Reply:
Because then I'd have to do
```
if isinstance(val, str):
    return val
else:  # enum
   return val.value
```
at each place where we access Rerankers/Generative/Vectorizer

With the `_EnumLikeStr` both have the same interface and we do not need to worry about that